### PR TITLE
MMRF: DM WGBS support and element-class generalization

### PR DESCRIPTION
**Pull request by compbiolover**
# Description
## Differential methylation at any regulatory element class

Differential methylation could only test one thing: the promoter matrix at `ds.queries.dnaMethylation.promoter`. This generalises it to arbitrary regulatory element classes, adds a WGBS-appropriate missingness rule, and makes element-level values readable per sample so a volcano hit can be inspected rather than just ranked.

Driven by the MMRF CoMMpass WGBS cohort (415 CD138+ samples), but nothing here is MMRF-specific — the dataset config lives in the sjpp PR.

### Element classes

`ds.queries.dnaMethylation` gains an `elements{}` map keyed by what the client sends as `element_type`. Several classes may share one HDF5 by declaring `element_class`, so offering cCRE promoters plus enhancer classes costs no extra files. The legacy `.promoter` key still works and remains the default when `element_type` is absent.

Two new config keys:

- `defaultElementType` — which class the volcano opens on. Without it, a dataset offering five classes still started on the legacy promoter matrix.
- `elementForTerms` — which matrix answers dnaMethylation *term* queries. A term carries coordinates but no element class, so a multi-class dataset has to nominate one.

**Hit counts are not comparable across classes.** The same contrast gives ~2,300 significant elements on ENCODE cCRE promoters and ~78,000 on TSS windows — the windows are wider and 15× more numerous. Downloaded p-value tables now record the element class in the run provenance line, alongside group sizes and every threshold, because a count without its class cannot be reproduced.

### Delta-beta is the default effect size for DM

The volcano now plots delta-beta for methylation and hides the log2 fold-change controls. `fold_change` on M-values is a difference of logits — it ranks elements correctly but says nothing interpretable about how much methylation changed, so a log2 cutoff would set a limit in units the plot never shows. The effect-size cutoff applies to delta-beta instead (default 0.1, a 10-percentage-point change). p-values still come from the limma fit on M-values.

### WGBS missingness

`platform: 'wgbs'` disables group-mean imputation. On arrays, NAs are structural — 450K, EPIC and EPIC v2 cover different probe sets — so imputing is reasonable. On WGBS a missing value only means that sample-element fell below the read-depth threshold, and imputing a group mean where the truth is "not sequenced deeply enough here" fabricates signal. limma fits each row on its finite observations instead.

### Element values readable per sample

A promoter-only WGBS cohort could run differential methylation but had no way to read one element's value for one sample — no violin by subgroup, no scatter against expression — even though the numbers were already in the element HDF5. `query_element_values.py` serves element matrices by coordinate range or element ID; `query_beta_values.py` cannot, since it needs a `chrom_lengths` attribute and `/meta/probe/probeID` that the element build doesn't write.

### Fixes

- Region-based dnaMethylation terms take their unit from the matrix serving them. An element matrix holding M-values was labelled "Average Beta Value", plotting values near −6 on an axis claiming a 0–1 scale.
- DA and scatter chart availability gate on the dnaMethylation **getter** rather than a CpG-level `.file`, so a dataset offering only element matrices is no longer configured-but-unreachable.
- `excludeSampleNamesMatching` withholds non-comparable specimen types at the single eligibility gate every DM path already filters through, and throws if the pattern matches nothing.
- `dnaMethylation` accepted as a `correlationVolcano` feature type.

### Tests

New: `promoterLabel.unit.spec.ts`, `VolcanoViewModel.unit.spec.ts`, `elementForTerms.unit.spec.ts`, `test_query_element_values.py`, plus diffMeth.R coverage in `Rscripts.spec.js`. Updated: `dnaMethylation.unit.spec.ts`, `sampleGroups.unit.spec.ts`, `termdb.config.unit.spec.ts`.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.

- [x] Tests: Added and/or passed unit and integration tests, or N/A
- [x] Todos: Commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A
- [x] Rust: Checked to see whether Rust needs to be re-compiled because of this PR, or N/A

### CODE REVIEW

File: client/tw/dnaMethylation.ts [hunk 1]
```diff
@@ -51,7 +51,11 @@ export function getDNAMethUnit(genomicFeatureType: string, vocabApi: VocabApi) {
 		case 'gene':
 			return vocabApi.termdbConfig.queries.dnaMethylation?.unit || 'Average Beta Value'
 		default:
-			return 'Average Beta Value'
+			/* A region term (the coordinate/gene-browser picker) is served by whatever matrix the
+			dataset nominates, which may hold M-values rather than betas -- so take the dataset's
+			declared unit and only fall back to beta when it declares none. Hardcoding beta here
+			mislabelled every region term on a WGBS element cohort. */
+			return vocabApi.termdbConfig.queries.dnaMethylation?.unit || 'Average Beta Value'
```
[the remark this answers is not in the document]
Comment (compbiolover) [reply client/tw/dnaMethylation.ts:58]: Already addressed in a previous commit to fix tests

File: shared/types/src/dataset.ts [hunk 2]
```diff
@@ -1280,17 +1280,101 @@ type Mds3Queries = {
...
+		 * This is a property of the data, not a preference, and it decides how missingness is
+		 * treated. On arrays a missing value is structural — 450K/EPIC/EPIC2 cover different
+		 * probe sets, so a sample is missing whole probes for reasons unrelated to its biology,
+		 * and imputing is the standard remedy. Under WGBS a missing value only means that
+		 * sample-region fell below the depth threshold, which is much rarer and closer to
+		 * random; imputing there adds no information and is slightly anticonservative. */
+		platform?: 'array' | 'wgbs'
+		/** promoter-by-sample matrix, values are average M-value.
+		 * Required for differential methylation (termdb/diffMeth). */
 		promoter?: {
```
Comment (xzhou82) [inline shared/types/src/dataset.ts:1304]: later may roll this under `dnaMethylation.elements.promoter`

**Review by xzhou82 — APPROVED** [review]
